File: minions/7.El hijo del medio/El_hijo_del_medio.c
#include <stdio.h>
#include <stdlib.h>

int main(){
    int a,b,c,medio;

    do{
        printf("Edades de los 3 hermanos? (cada una separada por un espacio)\n");
        scanf("%d %d %d",&a,&b,&c);
    } while (a<=0 || b<=0 || c<=0);

    if ((b<a && a<c) || (c<a && a<b)){
        medio = a;
    }else{
        if ((a<b && b<c) || (c<b && b<a)){
            medio = b;
        }else{
            if ((a<c && c<b) || (b<c && c<a)){
                medio = c;
            }
        }  
    }

    printf("el hijo del medio es : %i\n",medio);

    return EXIT_SUCCESS;
}

File: Ejercicios/07.El hijo del medio/El_hijo_del_medio.py
def main():
    a, b, c = 0, 0, 0
    
while True:
    try:
        a, b, c = map(int, input("Edades de los 3 hermanos?(separados por un espacio):").split())
            
        if a > 0 and b > 0 and c > 0:
            break
        else:
            print("ERROR, Todas las edades deben ser no negativas y distintas de 0.")
    except ValueError:
        print("ERROR, entrada no valida. Ingrese numeros enteros.")
            
if( b < a < c) or (c < a < b):
    medio=a
else:
    if( a < b < c) or (c < b < a):
        medio=b
    else:
        if( a < c < b) or ( b< c < a):
            medio = c

print ("El hijo del medio es {}".format(medio))
        
if __name__ == "__main__":
    main()